src/plugins/intel_gpu/src/runtime/ze/ze_device.cpp

```diff
@@ -0,0 +1,235 @@
+    info.max_image2d_height = device_image_properties.maxImageDims2D;
+
+    info.supports_fp16 = (device_module_properties.flags & ZE_DEVICE_MODULE_FLAG_FP16) != 0;
+    info.supports_fp64 = (device_module_properties.flags & ZE_DEVICE_MODULE_FLAG_FP64) != 0;
+    info.supports_fp16_denorms = info.supports_fp16 && (device_module_properties.fp16flags & ZE_DEVICE_FP_FLAG_DENORM) != 0;
+
+    info.supports_khr_subgroups = true;
+    info.supports_intel_subgroups = true;
+    info.supports_intel_subgroups_short = true;
+    info.supports_intel_subgroups_char = true;
+    info.supports_intel_required_subgroup_size = true;
+
+    info.supports_imad = (device_module_properties.flags & ZE_DEVICE_MODULE_FLAG_DP4A) != 0;
+    info.supports_immad = false; // FIXME
+
+    info.supports_usm = device_memory_access_properties.hostAllocCapabilities && device_memory_access_properties.deviceAllocCapabilities;
+
+    info.supports_local_block_io = true;
+
+    info.gfx_ver = {0, 0, 0}; // could find how to retrieve this from L0 so far
+    info.arch = gpu_arch::unknown;
+    info.ip_version = ip_version_properties.ipVersion;
+
+    info.device_id = device_properties.deviceId;
+    info.num_slices = device_properties.numSlices;
+    info.num_sub_slices_per_slice = device_properties.numSubslicesPerSlice;
+    info.num_eus_per_sub_slice = device_properties.numEUsPerSubslice;
+    info.num_threads_per_eu = device_properties.numThreadsPerEU;
+
+    info.num_ccs = compute_queue_props->numQueues;
+    info.supports_queue_families = true;
+
+    info.kernel_timestamp_valid_bits  = device_properties.kernelTimestampValidBits;
+    info.timer_resolution  = device_properties.timerResolution;
+
+    static_assert(ZE_MAX_DEVICE_UUID_SIZE == ov::device::UUID::MAX_UUID_SIZE, "");
+    static_assert(ZE_MAX_DEVICE_LUID_SIZE_EXT == ov::device::LUID::MAX_LUID_SIZE, "");
+    std::copy_n(&device_properties.uuid.id[0], ZE_MAX_DEVICE_UUID_SIZE, info.uuid.uuid.begin());
+
+    if (supports_luid) {
+        ze_device_luid_ext_properties_t luid_props{ZE_STRUCTURE_TYPE_DEVICE_LUID_EXT_PROPERTIES, nullptr};
+        ze_device_properties_t device_properties{ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES, &luid_props};
+        if (zeDeviceGetProperties(device, &device_properties) == ZE_RESULT_SUCCESS)
+            std::copy_n(&luid_props.luid.id[0], ZE_MAX_DEVICE_LUID_SIZE_EXT, info.luid.luid.begin());
+    }
+
+    info.supports_mutable_command_list = false;
+
+    if (supports_mutable_list) {
+        ze_mutable_command_list_exp_properties_t mutable_list_props = { ZE_STRUCTURE_TYPE_MUTABLE_COMMAND_LIST_EXP_PROPERTIES,  nullptr, 0, 0 };
+        ze_device_properties_t device_properties{ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES, &mutable_list_props};
+        if (zeDeviceGetProperties(device, &device_properties) == ZE_RESULT_SUCCESS) {
+            ze_mutable_command_exp_flags_t required_features = ZE_MUTABLE_COMMAND_EXP_FLAG_KERNEL_ARGUMENTS |
```

Comment:
1. you need verify which flags are support
2. do you want to capture everything?